operator/pkg/manifest/helm_processor.go

```diff
@@ -319,14 +324,68 @@ func (h *helm) renderManifestFromChartPath(chartPath string, flags types.Flags)
 	return release.Manifest, nil
 }
 
+// optimizedInstallCRDs is oriented on action.installCRDs, an internal method from HELM used for installing CRDs.
+// Usually, with a rendered release, this method gets called in front of the render to install dependencies before
+// resolving REST Mappings for kubernetes. However, since we do not use releases, but dry-run, we install CRDs
+// in advance.
+// There are 3 key differences between the method in the HELM client and this one
+// First, it resets the singleton client, allowing us to only reset the Mapper once per render process.
+// Second, it parses the manifest string based on the existing methods available in the HELM client.
+// Third, it creates crds and collects requirements concurrently to speed up CRD creation.
+func (h *helm) optimizedInstallCRDs(ctx context.Context, chartRequested *chart.Chart) error {
+	crds := chartRequested.CRDObjects()
+
+	// transform the stringified manifest into our resource list the same way we do for other resources
+	// but without any custom transforms
+	var crdManifest bytes.Buffer
+	for i := range crds {
+		crdManifest.Write(append(bytes.TrimPrefix(crds[i].File.Data, []byte("---\n")), '\n'))
+	}
+	resList, err := h.getTargetResources(ctx, crdManifest.String(), nil, nil)
+	if err != nil {
+		return err
+	}
+
+	crdInstallWaitGroup := sync.WaitGroup{}
+	errors := make(chan error, len(resList))
+	createCRD := func(i int) {
+		defer crdInstallWaitGroup.Done()
+		_, err := h.clients.KubeClient().Create(kube.ResourceList{resList[i]})
+		errors <- err
+	}
+
+	for i := range resList {
+		crdInstallWaitGroup.Add(1)
+		go createCRD(i)
+	}
+	crdInstallWaitGroup.Wait()
+	close(errors)
+
+	for err := range errors {
+		if err == nil || apierrors.IsAlreadyExists(err) {
+			continue
+		}
+		h.logger.Error(err, "failed on crd installation as pre-requisite for helm rendering")
+		return err
+	}
+
+	restMapper, err := h.clients.ToRESTMapper()
+	if err != nil {
+		return err
+	}
+	meta.MaybeResetRESTMapper(restMapper)
+
+	return nil
+}
+
 func (h *helm) setDefaultFlags(releaseName string) {
 	h.clients.Install().DryRun = true
 	h.clients.Install().Atomic = false
 
 	h.clients.Install().WaitForJobs = false
 
-	h.clients.Install().Replace = true     // Skip the name check
-	h.clients.Install().IncludeCRDs = true // include CRDs in the templated output
+	h.clients.Install().Replace = true // Skip the name check
+	h.clients.Install().IncludeCRDs = false
```

Comment:
I think we should persist CRDs in the persisted manifest, otherwise they wont be cleaned up on uninstall

Reply:
I think this should be done differently then. If you include CRDs you will also try to apply them with transformations and on every install. I suggest on uninstall we remove them explicitly.

Reply:
CRD flag should solve this problem.